util/registrar/src/registrar.rs

```diff
@@ -14,59 +14,41 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
 
-use futures::{Future, future, IntoFuture};
-use ethabi::{Address, Bytes};
-use std::sync::Arc;
+use call_contract::CallContract;
+use ethabi::Address;
 use keccak_hash::keccak;
+use types::ids::BlockId;
 
 use_contract!(registrar, "res/registrar.json");
 
 // Maps a domain name to an Ethereum address
 const DNS_A_RECORD: &'static str = "A";
 
-pub type Asynchronous = Box<dyn Future<Item=Bytes, Error=String> + Send>;
-pub type Synchronous = Result<Bytes, String>;
+/// Registrar contract interface
+/// Should execute transaction using current blockchain state.
+pub trait RegistrarClient: CallContract + Send + Sync {
+	/// Get address of the registrar itself
+	fn registrar_address(&self) -> Option<Address>;
 
-/// Registrar is dedicated interface to access the registrar contract
-/// which in turn generates an address when a client requests one
-pub struct Registrar {
-	client: Arc<dyn RegistrarClient<Call=Asynchronous>>,
-}
+	fn get_address<'a>(&self, key: &'a str, block: BlockId) -> Result<Option<Address>, String> {
```

Comment:
Missing docs?

Reply:
Added a doc line, and removed the line on the trait specifying that it's for the latest block.